020-Getting-started/110-solidstart.mdx

```diff
@@ -0,0 +1,830 @@
+        <p class="text-xl">{post?.description}</p>
+      </div>
+    </>
+  );
+}
+```
+
+Let's walk through what this does.
+
+Import `useParams` to enable access to the value of the `slug`. Import `useRouteData` and `createServerData$` to enable
+data to be loaded.
+
+```tsx
+import { Title } from 'solid-start';
+import { useParams, useRouteData } from 'solid-start';
+import { createServerData$ } from 'solid-start/server';
+```
+
+The `Title` component is imported from `solid-start` to allow the page title to be set. The `routeData` and
+`createServerData$` functions do the same job as the landing page.
+
+Instantiate a new `XataClient` instance, get the page parameters using the `useParams` hook, and then use the
+[`filter` function](https://xata.io/docs/typescript-client/get#the-typescript-sdk-functions-for-querying) on the
+auto-generated `Posts` property to perform a query on the Posts table and find the
+record where the `slug` column equals the value of `params.slug`. Use the
+[`getFirst` function](https://xata.io/docs/typescript-client/get#the-typescript-sdk-functions-for-querying) to access
+the first (and only) Post result and return the post as the `createServerData$` return value.
+
+```tsx
+import { XataClient } from '~/xata';
+
+export function routeData() {
+  return createServerData$(async () => {
+    const xata = new XataClient({
+      apiKey: import.meta.env.XATA_API_KEY,
+      branch: import.meta.env.XATA_BRANCH
+    });
+
+    const params = useParams();
+    return await xata.db.Posts.filter({ slug: params.slug }).getFirst();
+  });
+}
+```
+
+Finally, retrieve the `post` data using the `useLoaderData` hook. Then use the `post` data to set
+the page title via `<Title />` and update the rest of the page UI to contain all the information for the single Post:
+
+```tsx
+export default function Post() {
+  const data = useRouteData<typeof routeData>();
+  const post = data();
+
+  return (
+    <>
+      <Title>{post?.title}</Title>
+
+      <div class="w-full max-w-5xl mt-16">
+        <p class="mb-2">
+          <a href="/" class="text-purple-600">
+            &larr; Back to blog
+          </a>
+        </p>
+
+        <h1 class="text-3xl mb-2">{post?.title}</h1>
+        <p class="text-sm mb-4 text-purple-950 dark:text-purple-200">{post?.pubDate?.toDateString()}</p>
+        <p class="text-xl">{post?.description}</p>
+      </div>
+    </>
+  );
+}
+```
+
+The single blog post page will look as follows:
+
+![Single blog post](images/getting-started-blog-post.png)
+
+## Search posts
+
+The last piece of functionality to add to the application is full-text fuzzy search of blog posts.
+
+When you insert data into a Xata database, it is automatically indexed for full-text search. So you don't need to change
+any configuration to enable search, you just need to use the
+[TypeScript SDK search feature](https://xata.io/docs/typescript-client/search).
+
+Let's add this functionality to the landing page:
```

Comment:
```suggestion
Let's add this functionality to the index page:
```

I'd say index to reinforce the file name here?

Reply:
"landing" is used consistently throughout all the tutorials. I suggest leaving as-is. If we feel we should be more specific about using the name of the file rather than "landing", we should change it throughout.